src/app/core/core.state.ts

```diff
@@ -30,6 +39,33 @@ export const selectAuthState = createFeatureSelector<AppState, AuthState>(
   'auth'
 );
 
+export const selectRouterState = createFeatureSelector<
+  RouterReducerState<RouterStateUrl>
+>('router');
+
+@Injectable()
+export class CustomSerializer implements RouterStateSerializer<RouterStateUrl> {
```

Comment:
I think it will be cleaner if we move this to a separate file, thoughts?

Reply:
Yes indeed but I hope it will be remove soon as it should be the default serializer.

Reply:
`CustomSerializer` isn't the default serializer fron NgRx - https://github.com/ngrx/platform/blob/master/modules/router-store/src/serializer.ts#L23.


It isn't needed anymore to provide a custom serializer because the angular router snapshot became serializable, which was not the case before.

If we would remove the `CustomSerializer` we would have to do `route.firstChild.firstChild.firstChild.params.id` in our selector which isn't really pretty. So I would suggest to keep the `CustomSerializer`.